File: typescript/src/entities/Book.ts
import { ObjectId } from "mongodb";

export class Book {
    private id: ObjectId;
    private title: string;
    private author: string;
    private year: Date;
    private isbn: string;

    constructor(id: ObjectId, title: string, author: string, year: Date, isbn: string) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.year = year;
        this.isbn = isbn;
    }

    getId() {
        return this.id
    }
}

File: csharp/BookAPI/Entities/Book.cs
using BookAPI.Entities.Base;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace BookAPI.Entities
{
    public class Book : BaseEntity
    {
        public string title { get; set; }
        public string name { get; set; }
        public DateTime year { get; set; }
        [RegularExpression(@"^(?=(?:\D*\d){10}(?:(?:\D*\d){3})?$)[\d-]+$")]
        [DefaultValue("1248752418865")]
        public string isbn { get; set; }
    }

    public class BookDetailDto
    {
        public string title { get; set; }
        public string name { get; set; }
        public DateTime year { get; set; }
        [RegularExpression(@"^(?=(?:\D*\d){10}(?:(?:\D*\d){3})?$)[\d-]+$")]
        [DefaultValue("1248752418865")]
        public string? isbn { get; set; }
    }
}
